packages/manager/src/features/IAM/Shared/Permissions/Permissions.tsx

```diff
@@ -0,0 +1,105 @@
+import { StyledLinkButton, TooltipIcon } from '@linode/ui';
+import { debounce } from '@mui/material';
+import Grid from '@mui/material/Grid';
+import * as React from 'react';
+
+import {
+  StyledBox,
+  StyledClampedContent,
+  StyledContainer,
+  StyledGrid,
+  StyledListItem,
+  StyledSpan,
+  StyledTypography,
+  sxTooltipIcon,
+} from './Permissions.style';
+
+import type { PermissionType } from '@linode/api-v4/lib/iam/types';
+
+type Props = {
+  permissions: PermissionType[];
+};
+
+export const Permissions = ({ permissions }: Props) => {
+  const [showAll, setShowAll] = React.useState(false);
+  const [numHiddenItems, setNumHiddenItems] = React.useState<number>(0);
+
+  const containerRef = React.useRef<HTMLDivElement | null>(null);
+
+  const calculateHiddenItems = React.useCallback(() => {
+    if (showAll) {
+      setNumHiddenItems(0);
+      return;
+    }
+
+    const containerRect = containerRef.current?.getBoundingClientRect();
+    if (!containerRect) {
+      return;
+    }
+
+    const items = containerRef.current?.querySelectorAll('p');
+    if (!items) {
+      return;
+    }
+
+    const containerBottom = containerRect.top + containerRect.height;
+
+    const itemsArray = Array.from(items);
+    const firstHiddenIndex = itemsArray.findIndex(
+      (item: HTMLParagraphElement) => {
+        const rect = item.getBoundingClientRect();
+        return rect.top >= containerBottom;
+      }
+    );
+
+    const numHiddenItems = firstHiddenIndex
+      ? itemsArray.length - firstHiddenIndex
+      : 0;
+
+    setNumHiddenItems(numHiddenItems);
+  }, [showAll, permissions]);
+
+  React.useEffect(() => {
+    const handleResize = debounce(() => calculateHiddenItems(), 100);
+    calculateHiddenItems();
+    window.addEventListener('resize', handleResize);
+    return () => {
+      window.removeEventListener('resize', handleResize);
+    };
+  }, [calculateHiddenItems, showAll]);
+
+  return (
+    <Grid
+      container
+      data-testid="parent"
+      direction="column"
+      sx={{ marginBottom: 1 }}
+    >
+      <StyledGrid container item md={1}>
+        <StyledTypography variant="body1">Permissions</StyledTypography>
+        <TooltipIcon
+          status="help"
+          sxTooltipIcon={sxTooltipIcon}
+          text="Link is coming..."
+        />
+      </StyledGrid>
+
+      <StyledContainer>
+        <StyledClampedContent $showAll={showAll} ref={containerRef}>
+          <StyledBox>
+            {numHiddenItems > 0 && <StyledSpan> +{numHiddenItems} </StyledSpan>}
+            <StyledLinkButton onClick={() => setShowAll(!showAll)}>
+              {showAll ? 'Hide' : ` Expand`}
+            </StyledLinkButton>
+          </StyledBox>
+
+          {permissions.map((permission: string) => (
```

Comment:
Type safety: The permissions.map treats permissions as strings, but it's typed as `PermissionType[]. `

```suggestion
        {permissions.map((permission: PermissionType) => (
))}
```